core/domain/suggestion_services.py

```diff
@@ -1826,3 +2032,822 @@ def update_question_suggestion(
+            # hence final_reviewer_id should not be None when the suggestion is
+            # up-to-date.
+            suggestion_models.QuestionReviewStatsModel.get(
+                str(suggestion.final_reviewer_id), topic.topic_id
+            ))
+
+        if question_review_stat_model is None:
+            # This function is called when reviewing a question suggestion and
+            # hence final_reviewer_id should not be None when the suggestion is
+            # up-to-date.
+            suggestion_models.QuestionReviewStatsModel.create(
+                reviewer_user_id=str(suggestion.final_reviewer_id),
+                topic_id=topic.topic_id,
+                reviewed_questions_count=1,
+                accepted_questions_count=(1 if suggestion_is_accepted else 0),
+                accepted_questions_with_reviewer_edits_count=(
+                    1 if suggestion.edited_by_reviewer else 0),
+                first_contribution_date=suggestion.last_updated.date(),
+                last_contribution_date=suggestion.last_updated.date()
+            )
+            continue
+
+        question_review_stat = (
+            _create_question_review_stats_from_model(
+                question_review_stat_model))
+
+        increment_question_review_stats(
+            question_review_stat, suggestion.last_updated,
+            suggestion_is_accepted,
+            suggestion.edited_by_reviewer)
+        _update_question_review_stats_models([question_review_stat])
+
+    update_question_contribution_stats_at_review(suggestion)
+
+
+def increment_translation_contribution_stats_at_review(
+    translation_contribution_stat: (
+        suggestion_registry.TranslationContributionStats),
+    content_word_count: int,
+    suggestion_is_accepted: bool,
+    edited_by_reviewer: bool
+) -> None:
+    """Updates TranslationContributionStats object.
+
+    Args:
+        translation_contribution_stat: TranslationContributionStats. The stats
+            object to update.
+        content_word_count: int. The number of words in the translation.
+        suggestion_is_accepted: bool. A flag that indicates whether the
+            suggestion is accepted.
+        edited_by_reviewer: bool. A flag that indicates whether the suggestion
+            is edited by the reviewer.
+    """
+    translation_contribution_stat.accepted_translations_count += (
+        1 if suggestion_is_accepted else 0)
+    (
+        translation_contribution_stat
+        .accepted_translations_without_reviewer_edits_count
+    ) += (1 if edited_by_reviewer else 0)
+    translation_contribution_stat.accepted_translation_word_count += (
+       0 if not suggestion_is_accepted else content_word_count)
+    translation_contribution_stat.rejected_translations_count += (
+        1 if not suggestion_is_accepted else 0)
+    translation_contribution_stat.rejected_translation_word_count += (
+        0 if suggestion_is_accepted else content_word_count)
```

Comment:
As noted before, it still seems cleaner to me to increment within conditions than to have all these inline ternary conditions, e.g.

```
if suggestion_is_accepted:
  increment...
else:
  increment...

if edited_by_reviewer:
  increment...
```

Ditto for the other increment functions.

Reply:
Done